src/main/java/com/google/maps/DirectionsApiRequestBase.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2015 Google Inc. All rights reserved.
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+ * ANY KIND, either express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.google.maps;
+
+import com.google.maps.internal.ApiConfig;
+import com.google.maps.internal.ApiResponse;
+import com.google.maps.model.LatLng;
+import com.google.maps.model.TrafficModel;
+import com.google.maps.model.TransitMode;
+import com.google.maps.model.TransitRoutingPreference;
+import com.google.maps.model.TravelMode;
+import com.google.maps.model.Unit;
+import org.joda.time.ReadableInstant;
+
+import static com.google.maps.internal.StringJoin.join;
+
+/**
+ * Common base class for DirectionsApiRequest and FullDirectionsApiRequest.
```

Comment:
Sorry, it's really unclear to me how the different subclasses cause a different request to be sent up to the server. Can you explain it to me?